internal/db/db.go

```diff
@@ -2,281 +2,452 @@
 package db
 
 import (
+	"context"
 	"database/sql"
 	"errors"
 	"fmt"
 
+	"github.com/canonical/sqlair"
 	_ "github.com/mattn/go-sqlite3"
 	"golang.org/x/crypto/bcrypt"
 )
 
+// Database is the object used to communicate with the established repository.
+type Database struct {
+	certificateTable string
+	usersTable       string
+	conn             *sqlair.DB
+}
+
+type CertificateRequest struct {
+	ID int `db:"id"`
+
+	CSR              string `db:"csr"`
+	CertificateChain string `db:"certificate_chain"`
+	RequestStatus    string `db:"request_status"`
+}
+
+type User struct {
+	ID int `db:"id"`
+
+	Username       string `db:"username"`
+	HashedPassword string `db:"hashed_password"`
+	Permissions    int    `db:"permissions"`
+}
+
 const (
-	certificateRequestsTableName = "CertificateRequests"
+	certificateRequestsTableName = "certificate_requests"
 	usersTableName               = "users"
 )
 
-const queryCreateCSRsTable = `CREATE TABLE IF NOT EXISTS %s (
-	csr TEXT PRIMARY KEY UNIQUE NOT NULL, 
-	certificate TEXT DEFAULT ''
+const queryCreateCSRsTable = `
+	CREATE TABLE IF NOT EXISTS %s (
+	    id INTEGER PRIMARY KEY AUTOINCREMENT,
+
+		csr TEXT NOT NULL UNIQUE, 
+		certificate_chain TEXT DEFAULT '',
+		request_status TEXT DEFAULT 'Outstanding', 
+		
+		CHECK (request_status IN ('Outstanding', 'Rejected', 'Revoked', 'Active')),
+		CHECK (NOT (certificate_chain == '' AND request_status == 'Active' )),
+		CHECK (NOT (certificate_chain != '' AND request_status == 'Outstanding'))
+        CHECK (NOT (certificate_chain != '' AND request_status == 'Rejected'))
+        CHECK (NOT (certificate_chain != '' AND request_status == 'Revoked'))
 )`
 
-const (
-	queryGetAllCSRs = "SELECT rowid, * FROM %s"
-	queryGetCSR     = "SELECT rowid, * FROM %s WHERE rowid=?"
-	queryCreateCSR  = "INSERT INTO %s (csr) VALUES (?)"
-	queryUpdateCSR  = "UPDATE %s SET certificate=? WHERE rowid=?"
-	queryDeleteCSR  = "DELETE FROM %s WHERE rowid=?"
-)
+const queryCreateUsersTable = `
```

Comment:
(Unrelated to this PR - feel free to ignore) 

We'll likely want to relate users with certificate requests so that it's clear which user made which request